perun-core/src/test/java/cz/metacentrum/perun/core/entry/UsersManagerEntryIntegrationTest.java

```diff
@@ -390,22 +390,127 @@ public void updateUserWithNullValueInLastName() throws Exception {
 		User updateUser = usersManager.updateUser(sess, user);
 	}
 
-	@Test (expected=UserNotExistsException.class)
+	@Test
 	public void deleteUser() throws Exception {
 		System.out.println(CLASS_NAME + "deleteUser");
 
-		usersManager.deleteUser(sess, user, true);  // force delete
-		usersManager.getUserById(sess, user.getId());
-		// should be unable to get deleted user by his id
+		// Enable deletion of users
+		boolean originalUserDeletionForced = BeansUtils.getCoreConfig().getUserDeletionForced();
+		BeansUtils.getCoreConfig().setUserDeletionForced(true);
+
+		try {
+			usersManager.deleteUser(sess, user, true);  // force delete
+			usersManager.getUserById(sess, user.getId());
+			// should NOT be able to get deleted user by his id - should throw UserNotExistsException
+			fail();
+		} catch (UserNotExistsException ex) {
+			// this is expected
```

Comment:
After further thought, I believe having 'BeansUtils.getCoreConfig().setUserDeletionForced(true)' (and all the code that follows) inside the try block (like it is in ConsentsManager tests) is safer because it avoids missing other exceptions (which would result in the property not being set back to the original value). Please rewrite 'anonymizeUserWhenAnonymizationNotSupported()' (and the other tests you added ofc) as well.